test/extended/monitoring/collection_profiles.go

```diff
@@ -0,0 +1,219 @@
+					if !reflect.DeepEqual(selector, expectedSelector) {
+						return fmt.Errorf("expected selector %v, got %v", expectedSelector, selector)
+					}
+				}
+
+				return nil
+			})
+		})
+		g.It("should apply default collection profile to all managed monitors when enabled", func() {
+			monitors, err := r.fetchMonitorsFor( /* no profile */ )
+			o.Expect(err).To(o.BeNil())
+			o.Expect(monitors.Items).To(o.BeEmpty())
+			monitorCountPre := len(monitors.Items)
+
+			err = r.makeCollectionProfileConfigurationFor(tctx, collectionProfileFull)
+			o.Expect(err).To(o.BeNil())
+
+			o.Eventually(func() error {
+				monitors, err = r.fetchMonitorsFor([2]string{collectionProfileFeatureLabel, collectionProfileFull})
+				if err != nil {
+					return err
+				}
+				if len(monitors.Items) == 0 {
+					return fmt.Errorf("no monitors found with collection profile %q", collectionProfileFull)
+				}
+				monitorCountPost := len(monitors.Items)
+				if monitorCountPost != monitorCountPre {
+					return fmt.Errorf("expected %d monitors, got %d", monitorCountPre, monitorCountPost)
+				}
+
+				return nil
+			}).Should(o.BeNil())
+		})
+		g.It("should have at-least one implementation for each collection profile", func() {
+			for _, profile := range collectionProfilesSupportedList {
+				err = r.makeCollectionProfileConfigurationFor(tctx, profile)
+				o.Expect(err).To(o.BeNil())
+
+				o.Eventually(func() error {
+					monitors, err := r.fetchMonitorsFor([2]string{collectionProfileFeatureLabel, profile})
+					if err != nil {
+						return err
+					}
+					if len(monitors.Items) == 0 {
+						return fmt.Errorf("no monitors found with collection profile %q", profile)
+					}
+
+					return nil
+				}).Should(o.BeNil())
+			}
+		})
+	})
+})
+
+func (r *runner) fetchPrometheusSelectors() ([]*metav1.LabelSelector, error) {
+	var labelSelectors []*metav1.LabelSelector
+	p, err := r.mclient.Prometheuses(operatorNamespaceName).Get(tctx, "prometheus-k8s", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	labelSelectors = append(labelSelectors, p.Spec.ServiceMonitorSelector)
+	labelSelectors = append(labelSelectors, p.Spec.PodMonitorSelector)
+	labelSelectors = append(labelSelectors, p.Spec.ProbeSelector)
+	return labelSelectors, nil
+}
+
+func (r *runner) fetchMonitorsFor(selectors ...[2]string) (*pov1api.ServiceMonitorList, error) {
+	managedMonitorsSelectors := []string{
+		fmt.Sprintf("%s=%s", "app.kubernetes.io/part-of", operatorNamespaceName),
+	}
+	for _, selector := range selectors {
+		managedMonitorsSelectors = append(managedMonitorsSelectors, fmt.Sprintf("%s=%s", selector[0], selector[1]))
+	}
+	return r.mclient.ServiceMonitors(operatorNamespaceName).List(tctx, metav1.ListOptions{
+		LabelSelector: strings.Join(managedMonitorsSelectors, ","),
+	})
+}
+
+func (r *runner) makeCollectionProfileConfigurationFor(ctx context.Context, collectionProfile string) error {
+	// Create a configuration for the operator based on the current profile.
```

Comment:
The problem is that e2e tests usually come with a specific cluster monitoring configuration. Overwriting it isn't an option.

Reply:
So CMO's configuration is reconciled by some other operator and cannot be modified? Does this mean we cannot enable collection profiles at all (without hard-changing the configuration at source)?

Reply:
Perhaps those checks that require configuration changes would be better suited as part of the CMO tests themselves. 
Maybe we should only keep the ones that don't require modifying the configmap. For example, we could check that `profile:cluster_monitoring_operator_collection_profile:max` doesn't exist/empty or that a metric that could have been dropped by the minimal or full profiles is still present. If in the future, cProfiles is enabled by default with a profile being set, we can add more checks regarding that profile here.

Reply:
Yep, `profile:cluster_monitoring_operator_collection_profile:max` will exist in a `techpreview` cluster, and will be `0` at all times for all profiles (by default).

In the future, even if we did set a default profile, we would only ever have checks for that particular profile here, since switching won't be an option. This acts against the lower-bound requirement of [at-least five confident tests](https://github.com/openshift/api#defining-featuregate-e2e-tests) that should cover the feature in `o/origin` itself.

> If your FeatureGate lacks automated testing, there is an exception process that allows QE to sign off on the promotion by commenting on the PR.

It'd also make sense to add these in CMO instead, as you said, ask for an exception here, and have the QE signoff.

Reply:
The CMO config is defined in the test workflow after the cluster is installed but it's not modified afterwards (for now). To have the tests in origin, you'd need to make sure that the original CMO configuration is preserved. E.g. the collection profile tests only modify the `collectionProfile` field and keep all other content identical.